### ISSUE TITLE
react-ts template uses development bundle of React

### ISSUE THREAD

**Issue by christopher-caldwell** [vitejs/create-vite-app#47]
I apologize for making this an issue when it's more applicable for a google search, but I can't find anything about this.

## TL;DR

How do you build with the prod version of React?

When running the `vite build` command that comes with the `react-ts` template, React is bundled in development mode, causing a 267kb js file. 

## Reproduction Steps
- `yarn init vite-app ts-react --template react-ts`
- `cd ts-react`
- `yarn`
- `yarn build`
- Look at `dist/_assets/` and the size of the JS file. 

The file has
```
/** @license React v16.13.0
 * react-dom.development.js
 *
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
```

## What I've Tried
- Running `NODE_ENV=production && tsc && vite build` just to try to be sure. I see [this](https://github.com/vitejs/vite-plugin-react/blob/9031b924498a3121b2356c645a0084b899218c5b/src/resolver.ts#L5), but even with the setting, it bundles development.

**Comment by antfu** [comment]
Might be related to https://github.com/vitejs/vite/pull/732

**Comment by christopher-caldwell** [comment]
I added an env file with `export NODE_ENV='production'`, then changed the build command to `. env/.env.prod && tsc && vite build`.

 it worked as expected, with the production version of React. It's odd that setting the env during the command with `NODE_ENV=production && tsc && vite build` didn't work. I even added export before `NODE_ENV` to be sure I wasn't missing something.

If it's helpful here is [a demo repo](https://github.com/christopher-caldwell/vite-template-demo)

**Comment by yyx990803** [comment]
Should no longer be an issue with the latest template which uses official React packages.
